thriftrw/spec/struct.py

```diff
@@ -167,6 +167,22 @@ def from_wire(self, wire_value):
         # invalid.
         return self.surface(**kwargs)
 
+    def _from(self, value):
+        # check.type_code_matches(self, wire_value)
+        kwargs = {}
+        for field in self.fields:
+            field_value = getattr(value, field.name, None)
+            if field_value is None and hasattr(value, 'get'):
+                field_value = value.get(field.name)
+            if field_value is None:
+                continue
```

Comment:
This looks really inefficient. We should try to separate the `from_dict` and `__init__` behaviors.